src/backlight/backlight.c

```diff
@@ -509,10 +509,46 @@ static int build_save_file_path(sd_device *device, char **ret) {
         return 0;
 }
 
+static int device_new_from_arg(const char *s, sd_device **ret) {
+        _cleanup_(sd_device_unrefp) sd_device *device = NULL;
+        const char *subsystem, *sysname;
+        int r;
+
+        assert(s);
+        assert(ret);
+
+        sysname = strchr(s, ':');
+        if (!sysname)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Requires a subsystem and sysname pair specifying a backlight or LED device.");
+
+        subsystem = strndupa_safe(s, sysname - s);
```

Comment:
if you touch this, maybe switch this to strndup(). this is a caller controlled string, we should be careful with duplicating something unbounded caller controlled on the stack.